titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/common/V3QueueableTask.java

```diff
@@ -215,6 +215,10 @@ public AssignedResources getAssignedResources() {
         job.getJobDescriptor().getContainer().getHardConstraints().forEach((key, value) ->
                 constraintEvaluatorTransformer.hardConstraint(Pair.of(key, value), runningTasksGetter).ifPresent(result::add)
         );
+        if (null != constraintEvaluatorTransformer) {
```

Comment:
It is set to `null` by the integration test setup here (https://github.com/Netflix/titus-control-plane/blob/master/titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/scenario/JobsScenarioBuilder.java#L95). The `null` check is for these tests.

Reply:
For some reason, github isn't allowing me to reply to the above comment on the `Set<String> unassignedIpAddressIds`, so this is actually for that comment.

Yeah, this does not really need to be a concurrent structure since we always create new tasks for a job (and thus assign IPs) in the same thread's for() loop. This was mostly to handle cases if that were to change. I can change back to a non-concurrent data structure as it may be a needless protection.

Reply:
> It is set to `null` by the integration test setup here (snip). The `null` check is for these tests.

I'd recommend fixing the test to stop passing `null` in this case.

Reply:
Done.